backend/Squirrel.SqlService/Squirrel.SqlService.WebApi/Controllers/ConsoleAppHubController.cs

```diff
@@ -11,130 +12,165 @@ namespace Squirrel.SqlService.WebApi.Controllers;
 public class ConsoleAppHubController : ControllerBase
 {
     private readonly IHubContext<ConsoleAppHub, IExecuteOnClientSide> _hubContext;
+    private readonly ResultObserver _resultObserver;
 
-    public ConsoleAppHubController(IHubContext<ConsoleAppHub, IExecuteOnClientSide> hubContext)
+    public ConsoleAppHubController(IHubContext<ConsoleAppHub, IExecuteOnClientSide> hubContext, ResultObserver resultObserver)
     {
         _hubContext = hubContext;
+        _resultObserver = resultObserver;
     }
-
-
+    
+    private (Guid queryId, TaskCompletionSource<QueryResultTableDTO> tcs) RegisterQuery()
+    {
+        var queryId = Guid.NewGuid();
+        var tcs = _resultObserver.Register(queryId);
+        return (queryId, tcs);
+    }
+    
     // https://localhost:7244/api/ConsoleAppHub/getAllTablesNames
     [HttpPost("getAllTablesNames")]
     public async Task<ActionResult> GetAllTablesNamesAsync([FromBody] QueryParameters queryParameters)
     {
-        await _hubContext.Clients.User(queryParameters.ClientId).GetAllTablesNamesAsync(queryParameters.ClientId);
-        return NoContent();
+        var registerQuery = RegisterQuery();
```

Comment:
`var registerQuery = RegisterQuery();`

in each method but without sending data from `queryParameters`  - how does it work? 

(sorry for the stupid questions, a bit difficult for me to understand)

Reply:
queryParameters  used everywhere for sending data to console app? but yea there are some useless params for some method, I think we need create a couple DTOs but in another task, so RegisterQuery for register a separate request regardless of clientId(console guid) to distinguish them